src/librbd/cache/pwl/ssd/WriteLog.cc

```diff
@@ -666,6 +666,7 @@ bool WriteLog<I>::retire_entries(const unsigned long int frees_per_tx) {
             this->m_blocks_to_log_entries.remove_log_entry(write_entry);
           }
         }
+	first_valid_entry = control_block_pos + data_length + MIN_WRITE_ALLOC_SSD_SIZE;
```

Comment:
@MahatiC . it should:
first_valid_entry = sub_control_block_pos + data_length + MIN_WRITE_ALLOC_SSD_SIZE;
sub_control_block_pos = retiring_subentries.back()->log_entry_index.